kolibri/core/webpack/hooks.py

```diff
@@ -262,17 +262,20 @@ def js_and_css_tags(self):
     def frontend_message_tag(self):
         if self.frontend_messages():
             return [
-                '<script>{kolibri_name}.registerLanguageAssets("{bundle}", "{lang_code}", JSON.parse("{messages}"));</script>'.format(
+                """
+                <script>
+                    {kolibri_name}.registerLanguageAssets('{bundle}', '{lang_code}', JSON.parse({messages}));
+                </script>""".format(
                     kolibri_name="kolibriCoreAppGlobal",
                     bundle=self.unique_id,
                     lang_code=get_language(),
                     messages=json.dumps(
-                        self.frontend_messages(),
-                        separators=(",", ":"),
-                        ensure_ascii=False,
-                    )
-                    .replace("'", "\\'")
-                    .replace('"', '\\"'),
+                        json.dumps(
```

Comment:
At first I was confused as to why this works with only a singe `JSON.parse` on the frontend side of it.

Then I realized that the double dumps not only works, but is actually conceptually correct.

The first `json.dumps` just makes the Python object suitable for inlining in Javascript code as a valid Javascript object (previously we were kind of hacking that by putting it in quotation marks).

The second `json.dumps` takes the JSON string and makes is suitable for inlining in Javascript code as a valid Javascript string (which can then be properly fed to `JSON.parse`).

Reply:
Yeah, I enjoyed figuring this one out and the elegance of the solution

Reply:
If we wanted to be correct & safe everywhere, we should avoid

```python
"<script>console.log('{}')</script>".format(output_string)
```

and always do

```python
"<script>console.log({})</script>".format(json.dumps(output_string))
```

created a follow-up issue here: https://github.com/learningequality/kolibri/issues/6044